hera_sim/config/HERA_H1C_CONFIG.yaml

```diff
@@ -0,0 +1,66 @@
+foregrounds:
+    diffuse_foreground:
+        Tsky_mdl: 
+            datafile: HERA_Tsky_Reformatted.npz
+            interp_kwargs: {}
+        omega_p:
+            datafile: HERA_H1C_BEAM_POLY.npy
+            interp_kwargs:
+                object: beam
+                interpolator: poly1d
+io:
+    empty_uvdata:
+        start_freq: !!float 1e8
+        channel_width: 97656.25
+        integration_time: 10.7
+noise:
+    _get_hera_bm_poly:
+        datafile: HERA_H1C_BEAM_POLY.npy
+    resample_Tsky:
+        Tsky: 180.0
+        mfreq: 0.18
+        index: -2.5
+    sky_noise_jy:
+        inttime: 10.7
+    thermal_noise:
+        Tsky_mdl:
+            datafile: HERA_Tsky_Reformatted.npz
+            interp_kwargs: {}
+        omega_p:
+            datafile: HERA_H1C_BEAM_POLY.npy
+            interp_kwargs:
+                object: beam
+                interpolator: poly1d
+        Trx: 0
+        inttime: 10.7
+rfi:
+    _get_hera_stations:
+        datafile: HERA_H1C_RFI_STATIONS.npy
+    rfi_impulse:
+        chance: 0.001
+        strength: 20.0
+    rfi_scatter:
+        chance: 0.0001
+        strength: 10.0
+        std: 10.0
+    rfi_dtv:
+        freq_min: 0.174
+        freq_max: 0.214
+        width: 0.008
+        chance: 0.0001
+        strength: 10.0
+        strength_std: 10.0
+sigchain:
+    _get_hera_bandpass:
+        datafile: HERA_H1C_BANDPASS.npy
+    gen_bandpass:
+        gain_spread: 0.1
+    gen_whitenoise_xtalk:
+        amplitude: 3.0
+    gen_cross_coupling_xtalk:
+        amp: 0.0
+        dly: 0.0
+        phs: 0.0
+
+# XXX do we touch anything in utils, vis?
```

Comment:
I agree that 3 is probably the best, but it may be good to get input from other folks on this.

Reply:
I think if you and I agree, let's go with that. I guess the kinds of parameters that this definition excludes are those that are reasonable to have defaults, but don't really depend on a kind of instrument or analysis. These should just have defaults in the func sig, not in the YAML.

For example, defaults of 0.0 for parameters are signals that those parameters don't really have telescope-dependent defaults, but rather defaults that just "make sense".

Reply:
I'm thinking it may be good to modularize a bit; if we change the implementation to option (3), then it may be good to have the YAML configured something like:
```
instrument:
    beam: !beam
    ...
analysis:
    ...
```
What do you think?

Reply:
yes, agree!

Reply:
Regarding the `@component` and `@model` features, do you think it's a good idea to include these features in this PR? It seems like a somewhat independent feature from what I'm aiming to achieve in this PR, so I think it may be more appropriate to create another branch to implement these features.

Reply:
I feel like it's very much related, but perhaps you're right. We do want to be careful that we don't have too many PRs implementing _breaking_ changes though. If you think that implementing those will not break anything further, then definitely separate it.

Reply:
I suspect that I may be able to implement the `@model` and `@component` decorators without breaking anything, but it's hard to say without having actually begun development.

On another note, unless I'm missing some small details, the current state of this branch does not break any of the pre-existing API. I have been trying my best to be careful while developing and ensure that all of the code in `hera_sim` can be used in exactly the same way as it is currently used.